### PR TITLE
#21209: Added docs for cpu-idle and large messages

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

File: akka-docs/rst/scala/remoting-artery.rst [hunk 1]
```diff
@@ -402,7 +402,33 @@ Performance tuning
 Dedicated lane for large messages
```
Comment (patriknw) [inline akka-docs/rst/scala/remoting-artery.rst]: subchannel

File: akka-docs/rst/scala/remoting-artery.rst [hunk 2]
```diff
@@ -402,7 +402,33 @@ Performance tuning
...
+Akka remoting provides a dedicated channel for large messages if configured. Since actor message ordering must
+not be violated the channel is actually dedicated for *actors* instead of messages, to ensure all of the messages
+arrive in send order. It is possible to assign actors on given paths to use this dedicated channel by using
+path patterns:
+
+   akka.remote.artery.large-message-destinations = [
+      "/user/largeMessageActor",
+      "/user/largeMessagesGroup/*",
+      "/user/anotherGroup/*/largeMesssages",
+   ]
```
Comment (patriknw) [inline akka-docs/rst/scala/remoting-artery.rst]: also include a double wildcard example?

`"/user/thirdGroup/**"`

```
/user/thirdGroup/a/a1
/user/thirdGroup/a/a2
/user/thirdGroup/b/b1
```

File: akka-docs/rst/scala/remoting-artery.rst [hunk 3]
```diff
@@ -412,7 +438,19 @@ TODO
...
+
+  # Values can be from 1 to 10, where 10 strongly prefers latency
+  # and 1 strongly prefers less CPU usage
+  akka.remote.artery.advanced.idle-cpu-level = 1
+
+By setting this value to a lower number, it tells Akka to do longer "sleeping" periods on its thread dedicated
+for `spin-waiting <https://en.wikipedia.org/wiki/Busy_waiting>`_ and hence reducing CPU load when there is no
+immediate task to execute at the cost of a longer reaction time to an event when it actually happens. It is worth
+to be noted though that during a continuously high-throughput period this setting makes not much difference
+as the thread mostly has tasks to execute.
```
Comment (patriknw) [inline akka-docs/rst/scala/remoting-artery.rst]: Perhaps add something like:

This behavior is also the reason why you will see lower latency at a high message rate (10,000 messages/s) than at a low message rate (100 messages/s) as long as the system is not reaching saturation.

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-docs/rst/scala/remoting-artery.rst [hunk 4]
```diff
@@ -412,7 +441,20 @@ TODO
 Fine-tuning CPU usage latency tradeoff
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-TODO
+Artery has been designed for low latency and as a result it can be CPU hungry when the system is mostly idle.
+This is not always desirable. It is possible to tune the tradeoff between CPU usage and latency with
+the following configuration:
+
+  # Values can be from 1 to 10, where 10 strongly prefers latency
+  # and 1 strongly prefers less CPU usage
```
Comment (johanandren) [inline akka-docs/rst/scala/remoting-artery.rst]: strongly prefers _low_ latency

**Review by johanandren — APPROVED** [review]
LGTM except for one nitpick

File: akka-docs/rst/scala/remoting-artery.rst (hunk 4 above)
Comment (drewhk) [inline akka-docs/rst/scala/remoting-artery.rst]: Ok, I fix it in next round and I merge this.

**Pull request merged**